### PR TITLE
Add getTerrain method

### LINKED ISSUE #10278: add map.getTerrain()
All the other style spec properties setters on Map have corresponding getters. We should add map.getTerrain() which would return the serialized value previously set with setTerrain() or found in the stylesheet.

### PR DESCRIPTION
**Pull request by ryanhamley**
## Launch Checklist

 - [x] briefly describe the changes in this PR
    - closes #10278
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [x] manually test the debug page
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Add map.getTerrain() method</changelog>`

### CODE REVIEW

File: src/style/style.js [hunk 1]
```diff
@@ -1365,6 +1365,10 @@ class Style extends Evented {
         this.light.updateTransitions(parameters);
     }
 
+    getTerrain() {
+        return this.terrain ? this.terrain.get() : this.stylesheet.terrain;
```
Comment (karimnaaji) [inline src/style/style.js:1369]: I'm curious to why we return this.stylesheet.terrain when it's not set, the other root property getLight() does not adopt the same behavior. How do we detect whether terrain isn't set on a map?

Comment (ryanhamley) [reply src/style/style.js:1369]: This implementation was based off my understanding of this line from the original issue `We should add map.getTerrain() which would return the serialized value previously set with setTerrain() or found in the stylesheet`.

>How do we detect whether terrain isn't set on a map?
My understanding is that `this.terrain` would only be defined is `setTerrain` has been called. Is that not correct?

Comment (karimnaaji) [inline src/style/style.js:1369]: No that's unnecessary, we only need the following:

```suggestion
        return this.terrain ? this.terrain.get() : null;
```

If it's in the stylesheet, this.terrain will get itself initialized.

File: src/style/style.js [hunk 2]
```diff
@@ -1365,6 +1365,10 @@ class Style extends Evented {
         this.light.updateTransitions(parameters);
     }
 
+    getTerrain() {
```
Comment (karimnaaji) [inline src/style/style.js:1368]: Do we need this?

```suggestion
    getTerrain(): Terrain | null {
```

File: src/ui/map.js [hunk 3]
```diff
@@ -2149,6 +2149,15 @@ class Map extends Camera {
         return this._update(true);
     }
 
+    /**
+     * Returns the value of the terrain object.
+     *
+     * @returns {Object} terrain Terrain properties of the style.
+     */
+    getTerrain() {
```
Comment (karimnaaji) [inline src/ui/map.js:2157]: Same here

```suggestion
    getTerrain(): Terrain | null {
```

File: src/ui/map.js [hunk 4]
```diff
@@ -2149,6 +2149,15 @@ class Map extends Camera {
         return this._update(true);
     }
 
+    /**
+     * Returns the value of the terrain object.
```
Comment (karimnaaji) [inline src/ui/map.js:2153]:
```suggestion
     * Returns the terrain specification, or `null` if terrain isn't set on the map.
```

File: src/ui/map.js [hunk 5]
```diff
@@ -2149,6 +2149,15 @@ class Map extends Camera {
...
+     *
+     * @returns {Object} terrain Terrain properties of the style.
```
Comment (karimnaaji) [inline src/ui/map.js:2155]:
```suggestion
     * @returns {Object} terrain Terrain specification properties of the style.
```

**Review by karimnaaji — COMMENTED** [review]
Left a few suggestions/update to docs based on the first comment.

**Review by karimnaaji — APPROVED** [review]
Looks good!

**Pull request merged**